src/components/quiz/UserDetailsForm.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState } from "react";
+
+interface User {
+  name: string | null;
+  displayPic: string | null;
+}
+
+export const UserDetailsForm: React.FunctionComponent = ({status,setStatus}: any) => {
+
+  const [userDetails, setUserDetails] = useState<User>({
+    name: null,
+    displayPic:null,
+  });
```

Comment:
If we don't have a user logged in, then the entire `userDetails` object should be undefined, rather than having one with both fields set to `null`. An object with `null` fields implies "we do have a user logged in, but we know nothing about them"!

```suggestion
  const [userDetails, setUserDetails] = useState<User>();
```